pkg/nsx/services/securitypolicy/expand_test.go

```diff
@@ -236,3 +242,373 @@ func TestSecurityPolicyService_getPodSelectors(t *testing.T) {
 		})
 	}
 }
+
+var secPolicy = &v1alpha1.SecurityPolicy{
+	ObjectMeta: v1.ObjectMeta{Namespace: "ns1", Name: "p1", UID: "uid1"},
+	Spec: v1alpha1.SecurityPolicySpec{
+		Rules: []v1alpha1.SecurityPolicyRule{
+			{
+				Name:      "rule1",
+				Action:    &allowAction,
+				Direction: &directionIn,
+				Ports:     []v1alpha1.SecurityPolicyPort{},
+			},
+			{
+				Name:      "rule2",
+				Action:    &allowAction,
+				Direction: &directionIn,
+				Ports: []v1alpha1.SecurityPolicyPort{
+					{
+						Protocol: "TCP",
+						Port:     intstr.IntOrString{Type: intstr.Int, IntVal: 1000},
+					},
+					{
+						Protocol: "UDP",
+						Port:     intstr.IntOrString{Type: intstr.Int, IntVal: 1234},
+						EndPort:  1235,
+					},
+				},
+			},
+			{
+				Action:    &allowAction,
+				Direction: &directionIn,
+				Ports: []v1alpha1.SecurityPolicyPort{
+					{
+						Protocol: "TCP",
+						Port:     intstr.IntOrString{Type: intstr.String, StrVal: "http"}, // http port is 443
+					},
+					{
+						Protocol: "UDP",
+						Port:     intstr.IntOrString{Type: intstr.Int, IntVal: 1236},
+						EndPort:  1237,
+					},
+				},
+			},
+		},
+		AppliedTo: []v1alpha1.SecurityPolicyTarget{
+			{
+				PodSelector: &v1.LabelSelector{
+					MatchLabels: map[string]string{"pod_selector_1": "pod_value_1"},
+				},
+			},
+		},
+	},
+}
+
+func TestExpandRule(t *testing.T) {
+	ruleTagsFn := func(policyType string) []model.Tag {
+		return []model.Tag{
+			{Scope: common.String("nsx-op/cluster"), Tag: common.String("")},
+			{Scope: common.String("nsx-op/version"), Tag: common.String("1.0.0")},
+			{Scope: common.String("nsx-op/namespace"), Tag: common.String("ns1")},
+			{Scope: common.String("nsx-op/namespace_uid"), Tag: common.String("ns1-uid")},
+			{Scope: common.String(fmt.Sprintf("nsx-op/%s_name", policyType)), Tag: common.String("p1")},
+			{Scope: common.String(fmt.Sprintf("nsx-op/%s_uid", policyType)), Tag: common.String("uid1")},
+		}
+	}
+	npRuleTags := ruleTagsFn("network_policy")
+	spRuleTags := ruleTagsFn("security_policy_cr")
```

Comment:
for T1, we use 
"security_policy_cr_name" and "security_policy_cr_uid" tags
for VPC, we use
security_policy_name" and "security_policy_uid" tags
in our code, this is done by dynamically setting when initializing service, however, in our UT, we might need to
set it by our code itself to make UT results be consistent with our code logic.